src/Runner.hs

```diff
@@ -124,9 +124,11 @@ runModule repl (Module module_ setup examples) = do
   where
     reload :: IO ()
     reload = do
+#if __GLASGOW_HASKELL__ < 707
       -- NOTE: It is important to do the :reload first!  There was some odd bug
-      -- with a previous version of GHC (7.4.1?).
+      -- with a previous versions of GHC (7.6.3 and older).
       void $ Interpreter.safeEval repl   ":reload"
+#endif
       void $ Interpreter.safeEval repl $ ":m *" ++ module_
```

Comment:
Ah. Perhaps it's this bug? https://ghc.haskell.org/trac/ghc/ticket/5904

I see your point about needing to clear the scope between examples. It's frustrating that the only apparent method of doing so happens to be really slow...